challenge.c

```diff
@@ -142,15 +170,23 @@ typedef struct {
     uint16_t array[51];
 } q11_b_t;
 
-error_t q11(q11_a_t *a, q11_b_t *b){
+error_t q11(q11_a_t *a, q11_b_t *b) {
+    if (a == NULL || b == NULL) {
+        return FAIL;
+    }
+
+    size_t size_in_bytes = sizeof(a->array);
+    memcpy(&(b->array[1]), a->array, size_in_bytes - sizeof(uint16_t));
```

Comment:
`b->array` has a size of 51 `uint16_t`s and `a->array` has a size of 50 `uint16_t`s. You can copy `a->array` into `b->array` at an offset of 1 index without worrying about a buffer overflow. 

I.e. there's no need to subtract by `sizeof(uint16_t)`. By subtracting, you won't copy over all the contents of `a->array`. You'll miss the last `uint16_t`.